src/components/Calculator.js

```diff
@@ -1,41 +1,63 @@
 import React from 'react';
+import calculate from '../logic/calculate';
+import MagicBtn from './MagicBtn';
+import OutputResult from './OutputResult';
 
+const generateOutputValue = ({ total, next, operation }) => {
+  if (!total && !next && !operation) {
+    return '0';
+  } if (!total && next) {
+    return `${next}`;
+  } if (operation && total && !next) {
+    return `${total} ${operation}`;
+  } if (total && next && operation) {
+    return `${total} ${operation} ${next}`;
+  }
+  return `${total}`;
+};
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  clickEventListener = (event) => {
+    const object = this.state;
+    const { value } = event.target;
+    const result = calculate(object, value);
+    this.setState(result);
```

Comment:
**Changes Required**
- [ ] `Calculator.js` is a child component of `App.js`  Props need to get passed to child components of the Calculate.js component. You need to make `App.js` a stateful component and then pass down the state to it's children using `props`. In this case, Calculator.js should receive the state and `clickEventListener` as props defined in App.js

Reply:
- Noted we still have to refactor I guess that mkes more sense 👍